src/App.tsx

```diff
@@ -16,47 +19,92 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [sortValue, setSortValue] = useState(SortBY.none);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const sortBy = (value: string) => {
+    if (value === SortBY.alphabetically) {
+      if (isReversed) {
+        [...goodsFromServer].sort((a, b) => b.localeCompare(a));
+      } else {
+        [...goodsFromServer].sort((a, b) => a.localeCompare(b));
+      }
+
+      setSortValue(SortBY.alphabetically);
+
+      return;
+    }
+
+    if (value === SortBY.length) {
+      if (isReversed) {
+        [...goodsFromServer].sort((a, b) => b.length - a.length);
+      } else {
+        [...goodsFromServer].sort((a, b) => a.length - b.length);
+      }
+
+      setSortValue(SortBY.length);
+
+      return;
+    }
```

Comment:
let's improve you solution 
- make sense to use switch case statement here instead of many `if` blocks
- you need to apply reveres after sorting, for instance 

`goodsFromServer.sort()`
`if (isReversed) {`
`  goodsFromServer.reverse()`
`}`